### PR TITLE
Add type hinting for mongo provider

### PR DESCRIPTION
**Pull request by raffchen**
This PR adds type hinting for methods in the mongo provider per #9708
---
Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Target Github ISSUE in description if exists
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

File: airflow/providers/mongo/sensors/mongo.py [hunk 1]
```diff
@@ -41,13 +43,18 @@ class MongoSensor(BaseSensorOperator):
...
+                 mongo_conn_id: str = "mongo_default",
+                 *args,
+                 **kwargs) -> None:
         super().__init__(*args, **kwargs)
         self.mongo_conn_id = mongo_conn_id
         self.collection = collection
         self.query = query
 
-    def poke(self, context):
+    def poke(self, context: Optional[dict]) -> bool:
```
Comment (mik-laj) [inline airflow/providers/mongo/sensors/mongo.py]:
```suggestion
    def poke(self, context: dict) -> bool:
```
It's a required parameter.

**Review by mik-laj — APPROVED** [review]

**Comment by mik-laj** [comment]
@MagicTurtle2203  CI is sad. Can you fix it?

**Comment by MagicTurtle2203** [comment]
@mik-laj I've fixed it now :)

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Thanks @MagicTurtle2203 !

**Pull request merged**